validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java

```diff
@@ -387,19 +433,30 @@ protected SafeFuture<?> doStart() {
               validatorRestApi.ifPresent(restApi -> restApi.start().ifExceptionGetsHereRaiseABug());
               SystemSignalListener.registerReloadConfigListener(validatorLoader::loadValidators);
               validatorIndexProvider.lookupValidators();
-              eventChannels.subscribe(
-                  ValidatorTimingChannel.class,
-                  new ValidatorTimingActions(
-                      validatorStatusLogger,
-                      validatorIndexProvider,
-                      validatorTimingChannels,
-                      spec,
-                      metricsSystem));
-              validatorStatusLogger.printInitialValidatorStatuses().ifExceptionGetsHereRaiseABug();
-              return beaconNodeApi.subscribeToEvents();
+              if (maybeDoppelgangerDetectionService.isPresent()) {
+                return maybeDoppelgangerDetectionService
+                    .get()
+                    .start()
+                    .thenCompose(___ -> subscribeToEvents());
+              } else {
+                return subscribeToEvents();
+              }
             });
   }
 
+  private SafeFuture<Void> subscribeToEvents() {
+    eventChannels.subscribe(
+        ValidatorTimingChannel.class,
+        new ValidatorTimingActions(
+            validatorStatusLogger,
+            validatorIndexProvider,
+            validatorTimingChannels,
+            spec,
+            metricsSystem));
+    validatorStatusLogger.printInitialValidatorStatuses().ifExceptionGetsHereRaiseABug();
+    return beaconNodeApi.subscribeToEvents();
+  }
+
```

Comment:
```suggestion
              return maybeDoppelgangerDetectionService
                  .map(service -> service.start().toVoid())
                  .orElse(SafeFuture.COMPLETE);
            })
        .thenCompose(
            __ -> {
              eventChannels.subscribe(
                  ValidatorTimingChannel.class,
                  new ValidatorTimingActions(
                      validatorStatusLogger,
                      validatorIndexProvider,
                      validatorTimingChannels,
                      spec,
                      metricsSystem));
              validatorStatusLogger.printInitialValidatorStatuses().ifExceptionGetsHereRaiseABug();
              return beaconNodeApi.subscribeToEvents();
            });
```

I'd suggest a more compact\clean version

Reply:
Done